extensions/interactions/MathEquationInput/directives/math-equation-input-rules.service.ts

```diff
@@ -121,6 +122,95 @@ export class MathEquationInputRulesService {
     return false;
   }
+
+  ContainsSomeOf(
+      answer: MathEquationAnswer,
+      inputs: MathEquationRuleInputsWithSide): boolean {
+    let algebraicRulesService = new AlgebraicExpressionInputRulesService();
+
+    let positionOfTerms = inputs.y;
+
+    let splitAnswer = answer.split('=');
+    let lhsAnswer = splitAnswer[0], rhsAnswer = splitAnswer[1];
+
+    let splitInput = inputs.x.split('=');
+    let lhsInput = splitInput[0], rhsInput = splitInput[1];
+
+    if (positionOfTerms === 'lhs') {
+      return algebraicRulesService.ContainsSomeOf(lhsAnswer, {x: lhsInput});
+    } else if (positionOfTerms === 'rhs') {
+      return algebraicRulesService.ContainsSomeOf(rhsAnswer, {x: rhsInput});
+    } else if (positionOfTerms === 'both') {
+      return (
+        algebraicRulesService.ContainsSomeOf(lhsAnswer, {x: lhsInput}) && (
+          algebraicRulesService.ContainsSomeOf(rhsAnswer, {x: rhsInput})));
+    } else {
+      // Position of terms is irrelevant. So, we bring all terms on one side
+      // and perform an exact match.
+      let rhsAnswerModified = nerdamer(rhsAnswer).multiply('-1').text();
+      let expressionAnswer = nerdamer(rhsAnswerModified).add(lhsAnswer).text();
+
+      let rhsInputModified = nerdamer(rhsInput).multiply('-1').text();
+      let expressionInput = nerdamer(rhsInputModified).add(lhsInput).text();
+
+      return algebraicRulesService.ContainsSomeOf(
+        expressionAnswer, {x: expressionInput});
+    }
+  }
+
+  OmitsSomeOf(
+      answer: MathEquationAnswer,
+      inputs: MathEquationRuleInputsWithSide): boolean {
+    let algebraicRulesService = new AlgebraicExpressionInputRulesService();
+
+    let positionOfTerms = inputs.y;
+
+    let splitAnswer = answer.split('=');
+    let lhsAnswer = splitAnswer[0], rhsAnswer = splitAnswer[1];
+
+    let splitInput = inputs.x.split('=');
+    let lhsInput = splitInput[0], rhsInput = splitInput[1];
+
+    if (positionOfTerms === 'lhs') {
+      return algebraicRulesService.OmitsSomeOf(lhsAnswer, {x: lhsInput});
+    } else if (positionOfTerms === 'rhs') {
+      return algebraicRulesService.OmitsSomeOf(rhsAnswer, {x: rhsInput});
+    } else if (positionOfTerms === 'both') {
+      return (
+        algebraicRulesService.OmitsSomeOf(lhsAnswer, {x: lhsInput}) && (
+          algebraicRulesService.OmitsSomeOf(rhsAnswer, {x: rhsInput})));
+    } else {
+      // Position of terms is irrelevant. So, we bring all terms on one side
+      // and perform an exact match.
+      let rhsAnswerModified = nerdamer(rhsAnswer).multiply('-1').text();
+      let expressionAnswer = nerdamer(rhsAnswerModified).add(lhsAnswer).text();
+
+      let rhsInputModified = nerdamer(rhsInput).multiply('-1').text();
+      let expressionInput = nerdamer(rhsInputModified).add(lhsInput).text();
+
+      return algebraicRulesService.OmitsSomeOf(
+        expressionAnswer, {x: expressionInput});
+    }
+  }
+
+  MatchesWithGeneralForm(
```

Comment:
For this, do we allow any ordering around the = sign?

Reply:
No, this rule expects the learner's answer to match with the creator's answer on both sides of the equation. Since we want to check the general form match, it doesn't make sense in allowing the answer terms to be on either, since that would most likely always be the wrong answer. Do you agree?

Reply:
Hmm, ok, though a general form like ax+by+c=0 can be matched with 2x+3y=5, right, maybe? If we are not allowing this, might be worth adding a help text somewhere so the creator knows this.

Reply:
No, that wouldn't match. It's analogous to the "Matches exactly with on both sides" rule where LHS of the answer must match with the LHS of the input and same for RHS. Since if we allow this kind of behavior, y=mx+c would get matched with answers like 4x+y+3=0, which is not the expected behavior. I have added a help text for this rule as well.

Reply:
Yeah, agreed, thanks!